internal/provider/serverless_cluster_resource.go

```diff
@@ -0,0 +1,730 @@
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "serverless cluster resource",
+		Attributes: map[string]schema.Attribute{
+			"cluster_id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the cluster.",
+				Computed:            true,
+			},
+			"display_name": schema.StringAttribute{
+				MarkdownDescription: "The display name of the cluster.",
+				Required:            true,
+			},
+			"region": schema.SingleNestedAttribute{
+				MarkdownDescription: "The region of the cluster.",
+				Required:            true,
+				Attributes: map[string]schema.Attribute{
+					"name": schema.StringAttribute{
+						MarkdownDescription: "The unique name of the region.",
+						Required:            true,
+					},
+					"region_id": schema.StringAttribute{
+						MarkdownDescription: "The ID of the region.",
+						Computed:            true,
+					},
+					"cloud_provider": schema.StringAttribute{
+						MarkdownDescription: "The cloud provider of the region.",
+						Computed:            true,
+					},
+					"display_name": schema.StringAttribute{
+						MarkdownDescription: "The display name of the region.",
+						Computed:            true,
+					},
+				},
+			},
+			"spending_limit": schema.SingleNestedAttribute{
+				MarkdownDescription: "The spending limit of the cluster.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"monthly": schema.Int64Attribute{
+						MarkdownDescription: "Maximum monthly spending limit in USD cents.",
+						Optional:            true,
+						Computed:            true,
+					},
+				},
+			},
+			"automated_backup_policy": schema.SingleNestedAttribute{
+				MarkdownDescription: "The automated backup policy of the cluster.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"start_time": schema.StringAttribute{
+						MarkdownDescription: "The time of day when the automated backup will start.",
+						Optional:            true,
+					},
+					"retention_days": schema.Int64Attribute{
+						MarkdownDescription: "The number of days to retain automated backups.",
+						Optional:            true,
+					},
+				},
+			},
+			"endpoints": schema.SingleNestedAttribute{
+				MarkdownDescription: "The endpoints for connecting to the cluster.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.UseStateForUnknown(),
+				},
+				Attributes: map[string]schema.Attribute{
+					"public_endpoint": schema.SingleNestedAttribute{
+						MarkdownDescription: "The public endpoint for connecting to the cluster.",
+						Optional:            true,
```

Comment:
public_endpoint should not be Optional?

Reply:
the endpoints.public.disabled can be set